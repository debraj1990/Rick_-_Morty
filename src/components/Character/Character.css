.char{
    color:white;
    margin-top:2rem;
    border: 1px solid grey;
    border-radius: 0.5rem;
}
.cardStyle{
    /* width: 18rem; */
    color: white;
    background-color: transparent;
}
