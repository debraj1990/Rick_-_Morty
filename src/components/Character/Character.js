import React from "react";
import style from "./Character.css";



const Character = props => {
  // console.log("data-", props.data)

  if (!props.noResult) {
    return (<div className={style.char}>
      <div className={`${style.cardStyle} card`} >

        <img className="card-img-top"
          src={props.data.image}
          alt="name" />
        <div className="card-body" style={{ display: 'block', paddingLeft:' 0rem'}}>
          <h5 className="card-title">{props.data.name}</h5>
          <table className="table"
            style={{ fontSize: '1rem', tableLayout: 'fixed' }}>
            <tbody>
              <tr style={{ border: '0' }}>
                <th scope="col">ID:{props.data.id}</th>
                <th scope="col">Created {props.createdDate}</th>
              </tr>
              <tr>
                <th scope="col">SPECIES</th>
                <th scope="col">{props.data.species}</th>
              </tr>
              <tr>
                <th scope="col">STATUS</th>
                <th scope="col">{props.data.status}</th>
              </tr>
              <tr>
                <td>GENDER</td>
                <td>{props.data.gender}</td>
              </tr>
              <tr>
                <td>ORIGIN</td>
                <td>{props.data.origin.name}</td>
              </tr>
              <tr>
                <td>LAST LOCATION</td>
                <td>{props.data.location.name}</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>)
  } else {
    return (<div>
      <p>
        No results Found!
          </p>
    </div>)
  }


};


export default Character;