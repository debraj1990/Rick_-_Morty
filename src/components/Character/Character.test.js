import React from "react";
import { shallow } from 'enzyme';
import Character from './Character';


const mockData = {
    data: {
        "id": 1,
        "name": "Rick Sanchez",
        "status": "Alive",
        "species": "Human",
        "type": "",
        "gender": "Male",
        "origin": {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1"
        },
        "location": {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20"
        },
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    }
}
const mockCreatedDate = {
    createdDate: "3 years ago"
}
describe('character ', () => {
    it('Should Render character component ', () => {
        // given
        const wrapper = shallow(<Character data={mockData.data} />);
        // expected
        expect(wrapper).toMatchSnapshot();

    },
        it('should render name,id,createdDate,species,status,gender,origin,location props', () => {
            // given
            const wrapper = shallow(<Character data={mockData.data} createdDate={mockCreatedDate.createdDate} />);
            const name = wrapper.find('.card-title').html();
            const id = wrapper.find('.char-id').html();
            const createdDate = wrapper.find('.char-cdate').html();
            const species = wrapper.find('.char-species').html();
            const status = wrapper.find('.char-status').html();
            const gender = wrapper.find('.char-gender').html();
            const originName = wrapper.find('.char-origin').html();
            const locationName = wrapper.find('.char-loc').html();
            
            // expected
            expect(name).toContain(mockData.data.name);
            expect(id).toContain(mockData.data.id);
            expect(createdDate).toContain(createdDate);
            expect(species).toContain(mockData.data.species);
            expect(status).toContain(mockData.data.status);
            expect(gender).toContain(mockData.data.gender);
            expect(originName).toContain(mockData.data.origin.name);
            expect(locationName).toContain(mockData.data.location.name);

        })

    )
});




