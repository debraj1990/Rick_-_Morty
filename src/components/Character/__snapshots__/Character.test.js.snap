// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character  Should Render character component  1`] = `ShallowWrapper {}`;

exports[`character component Should Render  1`] = `ShallowWrapper {}`;

exports[`character component Should print  1`] = `ShallowWrapper {}`;
